test/conformance/program/program_adapter_opencl.match

```diff
@@ -1,3 +1,4 @@
+urProgramCreateWithILTest.BuildInvalidProgram/Intel_R__OpenCL___{{.*}}_
```

Comment:
I thought it did, but apparently the error was coming from `urProgramBuild` rather than create. I've added a quick test case for that.

Despite not passing in CI, this test does pass for me locally, so some devices handle it correctly.

If we aren't going to validate IR, should we amend the spec to explicitly call out that it's not required for implementations to validate that the IR is correct?

Reply:
Yeah it would be good to note that validation doesn't necessarily happen at the `ProgramCreate` step